server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreIT.java

```diff
@@ -301,16 +300,15 @@ public void testStaleCommitDeletionWithInvokeFlush() throws Exception {
         assertBusy(() -> {
             int actualFileCount = getFileCount(indexPath);
             if (numberOfIterations <= RemoteStoreRefreshListener.LAST_N_METADATA_FILES_TO_KEEP) {
-                assertEquals(numberOfIterations, actualFileCount);
+                assertTrue(numberOfIterations == actualFileCount || (numberOfIterations + 1) == actualFileCount);
```

Comment:
I think the following style will give you more details upon failure:

```
MatcherAssert.assertThat(actualFileCount, is(oneOf(numberOfIterations, numberOfIterations + 1)));
```

same applies below.

It looks like this still failed (on line 325), but the error message doesn't tell you what the actual value was, which I think the matcher style will do:

```
java.lang.AssertionError
	at __randomizedtesting.SeedInfo.seed([AAB48B28147F00DF:C606AA8AFF90B21D]:0)
	at org.junit.Assert.fail(Assert.java:87)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.junit.Assert.assertTrue(Assert.java:53)
	at org.opensearch.remotestore.RemoteStoreIT.testStaleCommitDeletionWithoutInvokeFlush(RemoteStoreIT.java:325)
```